### PR TITLE
Fix config init for non-writable locations

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

Fixes the case if the location isn't writable. The client should still work fine with built-in defaults.
```
➜  .config ls -l | grep kn
dr-xr-xr-x.  3 dsimansk dsimansk  4096 Oct 29 19:55 kn
```

```
➜  client git:(main) ✗ kn service list   
WARNING: failed writing config file to "/home/dsimansk/.config/kn/config.yaml": open /home/dsimansk/.config/kn/config.yaml: permission denied
Error: open /home/dsimansk/.config/kn/config.yaml: permission denied
Run 'kn --help' for usage
```

## Changes

* Fix config init for non-writable locations

## Reference

Follow-up #1472

### CODE REVIEW

File: pkg/kn/config/config.go [hunk 1]
```diff
@@ -145,10 +144,12 @@ func BootstrapConfig() error {
 			return fmt.Errorf("cannot stat configfile %s: %w", configFile, err)
 		}
 		if err := os.MkdirAll(filepath.Dir(viper.ConfigFileUsed()), 0775); err != nil {
-			return err
+			// Can't create config directory, proceed silently without reading the config
+			return nil
 		}
 		if err := os.WriteFile(viper.ConfigFileUsed(), []byte(configContentDefaults), 0600); err != nil {
-			fmt.Fprintf(os.Stderr, "WARNING: failed writing config file to %q: %s\n", configFile, err)
```
Comment (dsimansk) [inline pkg/kn/config/config.go:151]: @rhuss let me know if you want to go without any warning completely.

**Review by rhuss — APPROVED** [review]
/approve
/lgtm

**Review by boaz0 — APPROVED** [review]
LGTM.

Thanks @dsimansk

File: pkg/kn/config/config.go (hunk 1 above)
Comment (rhuss) [reply pkg/kn/config/config.go:151]: sounds good to me

**Pull request merged**